### PR TITLE
Akka 2.5.21 and type inference changes

### PR DESCRIPTION
**Pull request by ennru**
Use the latest Akka release.

Type inference changes in a few places.

### CODE REVIEW

File: mqtt-streaming/src/test/scala/docs/scaladsl/MqttSessionSpec.scala [hunk 1]
```diff
@@ -54,7 +54,7 @@ class MqttSessionSpec
               .clientSessionFlow(session)
               .join(pipeToServer)
           )
-          .toMat(Sink.collection)(Keep.both)
+          .toMat(Sink.seq)(Keep.both)
```
Comment (huntc) [inline mqtt-streaming/src/test/scala/docs/scaladsl/MqttSessionSpec.scala]: Why this change?

Comment (huntc) [reply mqtt-streaming/src/test/scala/docs/scaladsl/MqttSessionSpec.scala]: I see this issue remains also: https://github.com/akka/akka/issues/26305

Comment (ennru) [reply mqtt-streaming/src/test/scala/docs/scaladsl/MqttSessionSpec.scala]: Yes, that is the reason. I think this solution is the best for now.
